### ISSUE TITLE
Can't download any songs

### ISSUE THREAD

**Issue by pedrotrschneider** [jmshrv/finamp#325]
I'll use pictures from one specific example, but it works this way no matter what I try to download.
If I'm normally connected to my server and I select an album, it shows all information and plays the songs properly, like the image below:
![Screenshot_20220912-080442~2](https://user-images.githubusercontent.com/62211260/189642345-b42e4319-9629-4ff4-ab4a-ad13416b5d5e.png)
However, if I try to download the album by clicking the download button on the top right corner, the downloaded icon instantly appears on all songs (my internet is not nearly fast enough to download everything at these speeds) and the cover arts disappear, like in the image below:
![Screenshot_20220912-080556-302](https://user-images.githubusercontent.com/62211260/189642609-10097081-87d7-4cf3-80f5-4c9c84594569.png)
The downloads page tells me that all downloads finished successfully and nothing is left to be downloaded:
![Screenshot_20220912-080615-658](https://user-images.githubusercontent.com/62211260/189642758-722d8d85-e09f-491c-8685-38cc5f0b98b0.png)
At this point, I can't play any of the downloaded songs, be it in online mode or offline mode. The only way I can get the album to play again is by deleting the downloads and reloading the server. At this point, it functions normally again, but only online, of course.
Am I missing something or is this actually a bug?

Additional information:
- Device: Motorola Edge 20
- Software: Android 12, security patch from July 1, 2022
- The app was downloaded from the google play store

I'll gladly submit more information if requested.
Thanks in advance :)

**Comment by Chaphasilor** [comment]
Seems like a bug to me. One thing you could do is to close or force-stop the app (if you know how), then open it again, do the steps you described above, and then share the logs. Maybe there's a hint in there :)

![Screenshot_20220912-142339_Finamp.png](https://user-images.githubusercontent.com/18015147/189652555-da8f155c-2737-47e6-a04b-430349a56268.png)

**Comment by pedrotrschneider** [comment]
I was a bit reluctant to share the logs since I don't have the time now to go through it and check for sensitive information. I'll try to do it right now.

**Comment by pedrotrschneider** [comment]
So, one thing I just notice (I'm sorry for not mentioning this earlier, but I thought I checked for that) is that if I connect to the server using the LAN ip and port for my server (I host it on my local network), it works properly. However, I am using [ngrok](https://ngrok.com/) to port-forward my hosted server to the WAN and access it from any network.

Ngrok generates and hosts a link that forwards requests to my local server. The weird thing is: using it in on line mode works like a charm. The issue is only present when trying to download stuff. I took a look at the logs and saw some errors: flutter screaming about invalid image data and whatnot, but that was about it. This explains the images disappearing from downloaded content, but it doesn't explain why they don't play at all. Furthermore, how is it able to recognize the image data when connected to the internet, but now when downloading?

At any rate, I will be sharing the logs soon, I just need to clean them up and remove sensitive information.

**Comment by jmshrv** [comment]
Finamp sanitises logs for you so it should be fine. Always a good idea to double check though :)

**Comment by pedrotrschneider** [comment]
Oh, it's nice that it removes sensitive information! I was using the copy button like a dummy and didn't notice the share logs button.

As it turns out, the problem I was having was actually a problem with ngrok. I'll explain what I did to solve it and then close the issue, in case someone else runs into the same problem: I just stopped using ngrok :). What I did to access my server remotely was to setup a simple [ZeroTier](https://www.zerotier.com/) network and it worked perfectly. [This is a video](https://www.youtube.com/watch?v=EgdFzxit_6M) that explains how to set it up in detail in case someone has the same issue!

Thanks everyone for the fast response and sorry for posting an unrelated issue.
Also, just wanted to say, I'm loving the app! Really nice job :)
